File: 백준/Silver/1316. 그룹 단어 체커/그룹 단어 체커.py
import sys
input = sys.stdin.readline

n = int(input())
words = [input().rstrip() for _ in range(n)]
cnt = 0
for text in words:
    is_group = True
    for i in range(len(text) - 1):  
        if text[i] != text[i + 1]:
            if text[i + 1] in text[:i + 1]:
                is_group = False
                break
    if is_group:
        cnt += 1

print(cnt)

File: 백준/Silver/1316. 그룹 단어 체커/그룹 단어 체커.java
import java.io.*;
import java.util.*;

public class Main{
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        String [] words = new String[n];
        int cnt = 0;
        for(int i=0; i<n;i++){
            words[i] = sc.next();
        }
        for(String word : words){
            boolean is_group = true;
            for(int i=0;i<word.length()-1;i++){
                if (word.charAt(i) != word.charAt(i+1)){
                    if(word.indexOf(word.charAt(i+1),0)<i+1){
                        is_group = false;
                        break;
                    }
                }
            }
            if(is_group){
                cnt +=1;
            }
            
        }
        System.out.println(cnt);
        sc.close();
    }
}
